tests/cmaize/user_api/test_cmaize_option.cmake

```
@@ -0,0 +1,133 @@
+# Copyright 2023 CMakePP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(cmake_test/cmake_test)
+
+#[[[
+# Test the cmaize_option function.
+#]]
+ct_add_test(NAME "test_cmaize_option")
+function("${test_cmaize_option}")
+    include(cmaize/project/projects)
+    include(cmaize/user_api/cmaize_option)
+    include(cmaize/user_api/cmaize_project)
+
+    project("ATestCMaizeProject")
+    cmaize_project("ATestCMaizeProject")
```

To be safe, I would base the name of the project in a test off of the test name. I don't know if it will actually be an issue (I haven't looked into the inner workings of CMakeTest), but I worry about defining the same project name multiple times and having unwanted information pass over. I think I usually followed the pattern `<test_name>_project`.
```suggestion
    set(proj_name "test_cmaize_option_project")
    project("${proj_name}")
    cmaize_project("${proj_name}")
```